### PR TITLE
Long text spliter

### PR DESCRIPTION
**Pull request by ZHIHANCHEN03**
add `auto_split_long_text` parameter for TransformConfig and add using example notebook to use it.

### CODE REVIEW

File: uniflow/flow/client.py [hunk 1]
```diff
@@ -52,19 +59,76 @@ def __init__(self, config: TransformConfig) -> None:
...
         """
-        # convert config to dict for future remote calls
-        output = self._server.run(input_list)
+
+        processed_input = []
+
+        # Check if auto-splitting of long text is enabled
+        if self._config.auto_split_long_text:
+            # Define the token size limitation
+            token_size_limit = 4096
```
Comment (SayaZhang) [inline uniflow/flow/client.py:85]: nit: shall we move the parameter `token_size_limit` into the class init function?

File: uniflow/op/extract/split/recursive_character_splitter.py [hunk 2]
```diff
@@ -30,12 +33,23 @@ def __init__(
...
-            keep_separator: Whether to keep the separator in the chunks.
+            splitting_mode (str): "char" for character count, "token" for token count. Defaults to "char".
         """
         super().__init__(name)
         self._chunk_size = chunk_size
         self._chunk_overlap_size = chunk_overlap_size
         self._separators = separators or self.default_separators
+        self._splitting_mode = splitting_mode  # Track splitting mode
+        if self._splitting_mode == "token":
+            self._encoder = tiktoken.encoding_for_model(
```
Comment (SayaZhang) [inline uniflow/op/extract/split/recursive_character_splitter.py:44]: nit: `self._encoder` is only used once. So the `if` statement could be removed.

File: uniflow/flow/client.py (hunk 1 above)
Comment (ZHIHANCHEN03) [reply uniflow/flow/client.py]: Done

File: uniflow/op/extract/split/recursive_character_splitter.py (hunk 2 above)
Comment (ZHIHANCHEN03) [reply uniflow/op/extract/split/recursive_character_splitter.py]: Done

**Review by SayaZhang — APPROVED** [review]

**Pull request merged**